src/ModAttachr.ts

```diff
@@ -48,12 +55,18 @@ export class ModAttachr implements IModAttachr {
             for (const mod of settings.mods) {
                 this.addMod(mod);
             }
-        }
+        }
+
+        if (settings.eventNames) {
+          this.eventNames = settings.eventNames;
+        } else {
+          this.eventNames = new EventNames();
+        }

```

Comment:
This could be simplified into `this.eventNames = settings.eventNames || new EventNames();`.